src/Database/Firebase/Types.hs

```diff
@@ -20,6 +23,13 @@ data Firebase = Firebase {
 -- |Representation of a Firebase name response to a POST
 newtype Name = Name { unName :: String } deriving Show
 
+class ToLocation a where
+  toSegment :: a -> [String]
```

Comment:
I think in a previous commit you had this as `Text`; did you find that `String` was easier to use in practice?